app/src/main/java/org/thoughtcrime/securesms/ui/components/SessionOutlinedButton.kt

```diff
@@ -0,0 +1,204 @@
+package org.thoughtcrime.securesms.ui.components
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.material.ButtonColors
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.ButtonElevation
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collectLatest
+import network.loki.messenger.R
+import org.thoughtcrime.securesms.ui.LaunchedEffectAsync
+import org.thoughtcrime.securesms.ui.LocalPalette
+import org.thoughtcrime.securesms.ui.contentDescription
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+interface ButtonType {
+    @Composable fun border(color: Color, enabled: Boolean): BorderStroke?
+    @Composable fun buttonColors(color: Color): ButtonColors
+    val elevation: ButtonElevation? @Composable get
+
+    object Outline: ButtonType {
+        @Composable override fun border(color: Color, enabled: Boolean) = BorderStroke(1.dp, if (enabled) color else LocalPalette.current.disabled)
+        @Composable override fun buttonColors(color: Color) = ButtonDefaults.buttonColors(
+            contentColor = color,
+            backgroundColor = Color.Unspecified,
+            disabledContentColor = LocalPalette.current.disabled,
+            disabledBackgroundColor = Color.Unspecified
+        )
+        override val elevation: ButtonElevation? @Composable get() = null
+    }
+    object Fill: ButtonType {
+        @Composable override fun border(color: Color, enabled: Boolean) = null
+        @Composable override fun buttonColors(color: Color) = ButtonDefaults.buttonColors(
+            contentColor = LocalPalette.current.background,
+            backgroundColor = color,
+            disabledContentColor = LocalPalette.current.disabled,
+            disabledBackgroundColor = Color.Unspecified
+        )
+        override val elevation: ButtonElevation? @Composable get() = ButtonDefaults.elevation()
+    }
+}
+
+/**
+ * Base [Button] implementation
+ */
+@Composable
+fun Button(
```

Comment:
Isn't the name Button creating confusion with the material Button?
Could it be ButtonBase or something similar?

Reply:
Don't know if we're going to follow the google guidelines, but I found this:

> ### [Design, Usecase or Company/Project specific prefixes](https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-component-api-guidelines.md#name-of-a-component)
>  Avoid CompanyName (GoogleButton) or Module (WearButton) prefixes where possible and consider use-case or domain specific names if needed to. If the component you are building is a part of component library built using compose-foundation or compose-ui building blocks as a basis, the majority of the non-prefixed names should be available to developers without clashes: com.companyname.ui.Button or com.companyname.ui.Icon. Simple names make sure these components feel first-class when used.

Reply:
Yes, we can aim for similar guidelines.
This Button is still clashing with the Material Button.
`BasicButton` for example...